nflow-engine/src/main/java/io/nflow/engine/internal/guice/EngineModule.java

```diff
@@ -0,0 +1,212 @@
+        new AnnotationTransactionAttributeSource());
+    bindInterceptor(any(), annotatedWith(Transactional.class), transactionInterceptor);
+    install(new EngineInitModule());
+  }
+
+  static class GuiceDataSourceTransactionManager extends DataSourceTransactionManager {
+    private static final long serialVersionUID = 1L;
+
+    @Inject
+    public void setDatasource(@NFlow DataSource dataSource) {
+      super.setDataSource(dataSource);
+    }
+  }
+
+  @Provides
+  @Inject
+  @NFlow
+  @Singleton
+  public AbstractResource nflowNonSpringWorkflowsListing(Environment env) {
+    return engineConfiguration.nflowNonSpringWorkflowsListing(env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public ThreadFactory nflowThreadFactory() {
+    return engineConfiguration.nflowThreadFactory();
+  }
+
+  @Provides
+  @Inject
+  @Singleton
+  public WorkflowInstanceExecutor nflowExecutor(@NFlow ThreadFactory factory, Environment env) {
+    return engineConfiguration.nflowExecutor(factory, env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public ObjectMapper nflowObjectMapper() {
+    return engineConfiguration.nflowObjectMapper();
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public DataSource nflowDataSource(Environment env, DatabaseConfiguration databaseConfiguration) {
+    return databaseConfiguration.nflowDatasource(env, metricRegistry);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public DatabaseInitializer nflowDatabaseInitializer(@NFlow DataSource dataSource, Environment env,
+      DatabaseConfiguration databaseConfiguration) {
+    return databaseConfiguration.nflowDatabaseInitializer(dataSource, env);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public JdbcTemplate nflowJdbcTemplate(@NFlow DataSource dataSource, DatabaseConfiguration databaseConfiguration) {
+    return databaseConfiguration.nflowJdbcTemplate(dataSource);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public NamedParameterJdbcTemplate nflowNamedParameterJdbcTemplate(@NFlow DataSource dataSource,
+      DatabaseConfiguration databaseConfiguration) {
+    return databaseConfiguration.nflowNamedParameterJdbcTemplate(dataSource);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  @Inject
+  public TransactionTemplate nflowTransactionTemplate(@NFlow DataSourceTransactionManager txManager,
+      DatabaseConfiguration databaseConfiguration) {
+    return databaseConfiguration.nflowTransactionTemplate(txManager);
+  }
+
+  @Provides
+  @NFlow
+  @Singleton
+  public DataSourceTransactionManager nflowPlatformTransactionManager() {
+    return transactionManager;
+  }
+
+  @Provides
+  @Singleton
+  @Inject
+  public SQLVariants nflowSQLVariants(Environment env, DatabaseConfiguration databaseConfiguration) {
+    return databaseConfiguration.sqlVariants(env);
+  }
+
+  @Provides
+  @Singleton
+  @Inject
+  public DatabaseConfiguration databaseConfiguration(Environment env) {
```

Comment:
Maybe take `Collection<Class<SQLVariants>>` as parameter? and then loop through to find a match for the env?

Reply:
Decided to keep it like this